Packet++/src/IPv6Layer.cpp

```diff
@@ -258,6 +260,21 @@ void IPv6Layer::parseNextLayer()
 			? static_cast<Layer*>(new ESPLayer(payload, payloadLen, this, m_Packet))
 			: static_cast<Layer*>(new PayloadLayer(payload, payloadLen, this, m_Packet));
 		break;
+	case PACKETPP_IPPROTO_ICMPV6:
+	{
+		ProtocolType icmpv6Ver = IcmpV6Layer::getIcmpv6Version(payload, payloadLen);
+		if(icmpv6Ver == ICMPv6EchoRequest)
+			m_NextLayer = new ICMPv6EchoRequestLayer(payload, payloadLen, this, m_Packet);
+		else if(icmpv6Ver == ICMPv6EchoReply)
+			m_NextLayer = new ICMPv6EchoReplyLayer(payload, payloadLen, this, m_Packet);
+		else if(icmpv6Ver == NDPNeighborSolicitation)
+			m_NextLayer = new NDPNeighborSolicitationLayer(payload, payloadLen, this, m_Packet);
+		else if(icmpv6Ver == NDPNeighborAdvertisement)
+			m_NextLayer = new NDPNeighborAdvertisementLayer(payload, payloadLen, this, m_Packet);
```

Comment:
I think we can move this logic to `IcmpV6Layer` and have a static method `IcmpV6Layer::parseIcmpV6Message(...)` instead of `IcmpV6Layer::getIcmpv6Version()` that takes `payload`, `payloadLen`, `this` and `m_Packet` and creates the right layer.

We do something similar in BGP layer: https://github.com/seladb/PcapPlusPlus/blob/1a7e27c066d68d2c36f8c1f4c306fa15da1e6e7b/Packet%2B%2B/src/BgpLayer.cpp#L31